src/dra/5298C.c

```diff
@@ -2,7 +2,730 @@
 
 #define CH(x) ((x)-0x20)
 
-INCLUDE_ASM("dra/nonmatchings/5298C", func_800F298C);
+void func_800F298C(void) {
+    Entity* ent;
+    Primitive* prim;
+    LayerDef* layer;
+    s32 i;
+    s32 ent_unk68;
+    void (*unk_function)();
+    void (*unk_function2)();
+    s32 tempX;
+    s32 tempY;
+    s32* ptr_791c;
+    // This function is a state machine, this variable is some kind of
+    // overall state of the game engine
+    switch (D_8003C9A4) {
+    case 0:
+        if (g_IsUsingCd) {
+            return;
+        }
+        D_8006BB00 = 0;
+        D_801375C8 = 0;
+        D_8003C8B8 = 1;
+        g_StageId = func_800F16D0();
+        DestroyEntities(0);
+        DestroyAllPrimitives();
+        func_800EDAE4();
+        func_800EAD7C();
+        func_8010189C();
+        func_800F2404(0);
+        if ((g_StageId == STAGE_ST0) ||
+            (g_PlayableCharacter != PLAYER_ALUCARD)) {
+            unk_function = D_8013C004;
+            if (g_StageId == STAGE_ST0) {
+                unk_function(1);
```

Comment:
I'll leave renaming of functions for later since this PR is already making big changes in a lot of places. I believe I have addressed all your comments, but do let me know if I missed any. Thank you very much for helping to polish this!